docs/en/integrations/language-clients/java/index.md

```diff
@@ -5,126 +5,250 @@ keywords: [clickhouse, java, jdbc, client, integrate, r2dbc]
+|SimpleAggregateFunction|✔                    |✔                    |
+|AggregateFunction      |✗                    |✔                    |
 
-### Environment requirements
+:::note
+- AggregatedFunction - :warning: does not support `SELECT * FROM table ...`
+- Decimal - `SET output_format_decimal_trailing_zeros=1` in 21.9+ for consistency
+- Enum - can be treated as both string and integer
+- UInt64 - mapped to `long` in client-v1 
+:::
 
-- [OpenJDK](https://openjdk.java.net) version >= 8
+### Compatibility
+
+- All projects in this repo are tested with all [active LTS versions](https://github.com/ClickHouse/ClickHouse/pulls?q=is%3Aopen+is%3Apr+label%3Arelease) of ClickHouse.
+- [Support policy](https://github.com/ClickHouse/ClickHouse/blob/master/SECURITY.md#security-change-log-and-support)
+- We recommend to upgrade client continuously to not miss security fixes and new improvements
+  - If you have an issue with migration - create and issue and we will respond! 
 
-### Compatibility with ClickHouse
+## Client-V2
 
-| Client version | ClickHouse |
-|----------------| ---------- |
-| 0.6.0          | 22.8+      |
+Implementation of a new API. It uses Apache Http Client to communicate with ClickHouse server. We have selected this http client because it has many built-in features and 
+has proven itself in old client implementation. We are planning to support other http client libraries. 
 
-### Installation
+*Note*: Client-V2 is currently in the phase of active development and we are still working on it. 
+
+### Setup
+
+- Maven Central web page: https://mvnrepository.com/artifact/com.clickhouse/client-v2
+- Nightly builds repository: https://s01.oss.sonatype.org/content/repositories/snapshots/com/clickhouse/
 
 ```xml
+<!-- https://mvnrepository.com/artifact/com.clickhouse/client-v2 -->
 <dependency>
     <groupId>com.clickhouse</groupId>
-    <!-- or clickhouse-grpc-client if you prefer gRPC -->
-    <artifactId>clickhouse-http-client</artifactId>
-    <version>0.6.0</version>
+    <artifactId>client-v2</artifactId>
+    <version>0.6.5</version>
 </dependency>
+
 ```
 
-Since version `0.5.0`, the driver uses a new client http library that needs to be added as a dependency.
+### Initialization
+
+Client object is initialized by `com.clickhouse.client.api.Client.Builder#build()`. Each client has own context and no objects are shared between them.
+Builder has configuration method for convinient setup. 
+
+Example: 
+```java
+
+ Client client = new Client.Builder()
+                .addEndpoint("https://clickhouse-cloud-instance:8443/")
+                .setUsername(user)
+                .setPassword(password)
+                .build;
 
-```xml
-<dependency>
-   <groupId>org.apache.httpcomponents.client5</groupId>
-   <artifactId>httpclient5</artifactId>
-   <version>5.2.3</version>
-</dependency>
 ```
 
-### Supported data types
+`Client` is `AutoCloseable` and should be closed when not needed anymore. 
```

Comment:
It's quite an important change. Should we move above the code snippet?

Reply:
Added to according sections in client v2 (see QueryResponse)